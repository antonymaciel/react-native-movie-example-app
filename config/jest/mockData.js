export const movie = {
    id: 0,
    year: '2019',
    imageUrl: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    title: 'Spiderman',
    vote_average: '9.1',
    urlbackdropImage: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg', 
    generes: 'Aventure, Drama, Terror',  
    vote_count: '1200',
    overview: 'Amazing movie',
    backdrop_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    genres: ['Adventure', 'Terror', 'Drama'],
    release_date: '2016-5-14'
}

export const review = {
    id: 0,
    author: 'Antony Maciel',
    url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
    content: 'Great Movie!, always has amazing content \n I love this movie'
}

export const movies = [
    {
        id: 1,
        poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'XMan',
        vote_average: 7.8,
        release_date: '2016-5-14'
    },
    {     
        id: 2,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Superman',
        vote_average: 0.9,
        release_date: '2015-5-9'
    },
    {
        id: 3,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Spiderman',
        vote_average: 8.0,
        release_date: '2005-5-27'
    }
]


export const reviews = [
    {
        id: 1,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!!!'
    },
    {
        id: 2,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!, always has amazing content \n I love this movie'
    },
    {
        id: 3,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: '&^%$#@!'
    }
]

