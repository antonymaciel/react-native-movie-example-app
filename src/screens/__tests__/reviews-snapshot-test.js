import React from 'react';
import renderer from 'react-test-renderer';
import { reviews, movie } from '../../../config/jest/mockData';
import Reviews from '../reviews';


it('renders home screen using Snapshots', () => {
  expect(renderer.create(
    <Reviews
        total_results={3}
        results={reviews}
        posterUrl={movie.imageUrl}
        onNewPage={jest.fn()} 
    />
  )).toMatchSnapshot();
});
