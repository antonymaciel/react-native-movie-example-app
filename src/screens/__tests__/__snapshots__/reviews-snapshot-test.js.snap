// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home screen using Snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "500",
        }
      }
    >
      Reviews (
      3
      )
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
        }
      }
      style={
        Object {
          "height": 130,
          "marginRight": 40,
          "width": 90,
        }
      }
    />
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListFooterComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "author": "Antony Maciel",
          "content": "Great Movie!!!",
          "id": 1,
          "url": "https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df",
        },
        Object {
          "author": "Antony Maciel",
          "content": "Great Movie!, always has amazing content 
 I love this movie",
          "id": 2,
          "url": "https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df",
        },
        Object {
          "author": "Antony Maciel",
          "content": "&^%$#@!",
          "id": 3,
          "url": "https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyboardShouldPersistTaps="always"
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.2}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <ReviewComponent
          author="Antony Maciel"
          content="Great Movie!!!"
          url="https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df"
        />
        <View />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <ReviewComponent
          author="Antony Maciel"
          content="Great Movie!, always has amazing content 
 I love this movie"
          url="https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df"
        />
        <View />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <ReviewComponent
          author="Antony Maciel"
          content="&^%$#@!"
          url="https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df"
        />
      </View>
      <View
        onLayout={[Function]}
      >
        <View />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
